src/physics/navier_stokes.cpp

```diff
@@ -956,13 +1417,406 @@ ::boundary_manufactured_solution (
+        this->pcout << " Did not assign a value to all the data. Missing " << computed_quantities.size() - current_data_index << " variables."
+                  << " If you added a new output variable, make sure the names and DataComponentInterpretation match the above. "
+                  << std::endl;
+    }
+
+    return computed_quantities;
+}
+
+template <int dim, int nstate, typename real>
+std::vector<dealii::DataComponentInterpretation::DataComponentInterpretation> NavierStokes<dim,nstate,real>
+::post_get_data_component_interpretation () const
+{
+    namespace DCI = dealii::DataComponentInterpretation;
+    std::vector<DCI::DataComponentInterpretation> interpretation = PhysicsBase<dim,nstate,real>::post_get_data_component_interpretation (); // state variables
+    interpretation.push_back (DCI::component_is_scalar); // Density
+    for (unsigned int d=0; d<dim; ++d) {
+        interpretation.push_back (DCI::component_is_part_of_vector); // Velocity
+    }
+    for (unsigned int d=0; d<dim; ++d) {
+        interpretation.push_back (DCI::component_is_part_of_vector); // Momentum
+    }
+    interpretation.push_back (DCI::component_is_scalar); // Total Energy
+    interpretation.push_back (DCI::component_is_scalar); // Pressure
+    interpretation.push_back (DCI::component_is_scalar); // Pressure coefficient
+    interpretation.push_back (DCI::component_is_scalar); // Temperature
+    interpretation.push_back (DCI::component_is_scalar); // Entropy generation
+    interpretation.push_back (DCI::component_is_scalar); // Mach number
+    if constexpr(dim==3) {
+        for (unsigned int d=0; d<3; ++d) {
+            interpretation.push_back (DCI::component_is_part_of_vector); // Vorticity
+        }
+    }
+    interpretation.push_back (DCI::component_is_scalar); // Vorticity magnitude
+    interpretation.push_back (DCI::component_is_scalar); // Enstrophy
+    interpretation.push_back (DCI::component_is_scalar); // Second-invariant Q
+    interpretation.push_back (DCI::component_is_scalar); // Dilatation
+    interpretation.push_back (DCI::component_is_scalar); // Density gradient magnitude
+
+    std::vector<std::string> names = post_get_names();
+    if (names.size() != interpretation.size()) {
+        this->pcout << "Number of DataComponentInterpretation is not the same as number of names for output file" << std::endl;
+    }
+    return interpretation;
+}
+
+
+template <int dim, int nstate, typename real>
+std::vector<std::string> NavierStokes<dim,nstate,real>
+::post_get_names () const
+{
+    std::vector<std::string> names = PhysicsBase<dim,nstate,real>::post_get_names ();
+    names.push_back ("density");
+    for (unsigned int d=0; d<dim; ++d) {
+      names.push_back ("velocity");
+    }
+    for (unsigned int d=0; d<dim; ++d) {
+      names.push_back ("momentum");
+    }
+    names.push_back ("total_energy");
+    names.push_back ("pressure");
+    names.push_back ("pressure_coeffcient");
```

Comment:
this could be an opportunity to correct this typo (`pressure_coefficient`). But I'm not sure if that would mess up anyone's paraview scripts...

Reply:
Sure I can fix it; doesn't break the code so why not